lib/widgets/path_editor/editors/graph_editor.dart

```diff
@@ -0,0 +1,427 @@
+    }
+
+    ExtraLinesData extraLines = ExtraLinesData(horizontalLines: [
+      HorizontalLine(
+        y: 0,
+        color: Colors.white60,
+        strokeWidth: 3,
+      ),
+    ], verticalLines: markers, extraLinesOnTop: false);
+
+    SideTitles leftTitles = SideTitles(
+      getTitlesWidget: leftTitleWidgets,
+      showTitles: true,
+      reservedSize: 40,
+    );
+
+    SideTitles bottomTitles = SideTitles(
+      showTitles: true,
+      reservedSize: 32,
+      getTitlesWidget: bottomTitleWidgets,
+    );
+
+    FlTitlesData titlesData = FlTitlesData(
+      bottomTitles: AxisTitles(
+        sideTitles: bottomTitles,
+      ),
+      rightTitles: AxisTitles(
+        sideTitles: SideTitles(showTitles: false),
+      ),
+      topTitles: AxisTitles(
+        sideTitles: SideTitles(showTitles: false),
+      ),
+      leftTitles: AxisTitles(
+        sideTitles: leftTitles,
+      ),
+    );
+
+    FlBorderData borderData = FlBorderData(
+      show: true,
+      border: const Border(
+        bottom: BorderSide(color: Colors.white24, width: 4),
+        left: BorderSide(color: Colors.white24, width: 4),
+        right: BorderSide(color: Colors.white24, width: 4),
+        top: BorderSide(color: Colors.white24, width: 4),
+      ),
+    );
+
+    LineChartData data = LineChartData(
+      lineTouchData: LineTouchData(enabled: false),
+      gridData: FlGridData(show: true),
+      titlesData: titlesData,
+      borderData: borderData,
+      lineBarsData: lineBarsData,
+      extraLinesData: extraLines,
+      minX: 0,
+      maxX: time,
+      minY: min.floorToDouble(),
+      maxY: max.ceilToDouble(),
+    );
+
+    return LineChart(
+      data,
+      swapAnimationDuration: const Duration(milliseconds: 0),
+    );
+  }
+
+  String markerLabel(VerticalLine line) {
+    for (int idx = 0; idx < path.markers.length; idx++) {
+      if (path.markers[idx].timeSeconds == line.x) {
+        return path.markers[idx].name;
+      }
+    }
+    return '';
+  }
+
+  Widget leftTitleWidgets(double value, TitleMeta meta) {
+    if ((value - value.toInt()).abs() > 0.01) {
+      return const Text('');
+    }
+
+    const style = TextStyle(
+      color: Colors.white,
+      fontWeight: FontWeight.bold,
+      fontSize: 14,
+    );
+
+    return SideTitleWidget(
+        axisSide: meta.axisSide,
+        space: 10,
+        child: Text(value.toInt().toString(),
+            style: style, textAlign: TextAlign.center));
+  }
+
+  Widget bottomTitleWidgets(double value, TitleMeta meta) {
+    if (((value * 10.0) - (value * 10.0).round().toInt()).abs() > 0.01) {
+      return const Text('');
+    }
+
+    const style = TextStyle(
+      color: Colors.white,
+      fontWeight: FontWeight.bold,
+      fontSize: 16,
+    );
+
+    return SideTitleWidget(
+      axisSide: meta.axisSide,
+      space: 10,
+      child: Text(value.toStringAsFixed(1), style: style),
+    );
+  }
+}
+
+class GraphEditor extends StatefulWidget {
```

Comment:
I'd move this class above the `_LineChart` class so the main widget class for this file is at the top